ewfs/ewfs.py

```diff
@@ -298,33 +269,70 @@ def _ewfs_rotation(self, qc: QuantumCircuit, observer: int, angle: float, invert
-                bob_size = 1
+        for bitstring, count in results[setting].items():
+            processed_bitstring = bitstring.replace(" ", "")
+
+            flags_bitstring = processed_bitstring[:flag_size]
+            friends_bitstring = processed_bitstring[flag_size:]
+
+            if flags_bitstring == "0" * flag_size:
+                post_selected_results_setting[friends_bitstring] = count
+
+        post_selected_results[setting] = post_selected_results_setting
+    return post_selected_results
+
+def decode_results(post_selected_results, charlie_size) -> dict:
+    """Take majority vote of measurement bit-strings."""
+    decoded_results = {}
+
+    # For each setting, there is a dictionary of measurement results.
+    for setting in post_selected_results:
+        if setting[0] == "peek":
+            # Debbie's size is 1 because no PEEK setting
+            bob_size = 1
+
+            setting_results: dict = {}
+            # Decode the keys for each measurement result of the setting.
+            for k, v in post_selected_results[setting].items():
+                alice_result, bob_result = k[-charlie_size :], k[:bob_size]
+
+                alice_zero_count, bob_zero_count = alice_result.count("0"), bob_result.count("0")
+
+                alice_decoding = "0" if alice_zero_count >= charlie_size // 2 + 1 else "1"
+                bob_decoding = "0" if bob_zero_count >= 1 else "1"
+
+                if alice_decoding + bob_decoding in setting_results.keys():
+                    setting_results[alice_decoding + bob_decoding] += v
+                else:
+                    setting_results[alice_decoding + bob_decoding] = v
+            decoded_results[setting] = setting_results
+        else:
+            decoded_results[setting] = post_selected_results[setting]
 
-                setting_results: dict = {}
-                # Decode the keys for each measurement result of the setting.
-                for k, v in results[setting].items():
-                    alice_result, bob_result = k[-self.charlie_size :], k[:bob_size]
+    return decoded_results
 
-                    alice_zero_count, bob_zero_count = alice_result.count("0"), bob_result.count("0")
+def double_expect(settings: tuple[str, str], results: dict) -> float:
+    """Expectation value of product of two operators."""
+    probs = results[settings]
+    # <AB> = P(00) - P(01) - P(10) + P(11)
+    return probs.get("00", 0.0) - probs.get("01", 0.0) - probs.get("10", 0.0) + probs.get("11", 0.0)
 
-                    alice_decoding = "0" if alice_zero_count >= self.charlie_size // 2 + 1 else "1"
-                    bob_decoding = "0" if bob_zero_count >= 1 else "1"
+def semi_brukner_value(probs):
+    
+    A1B2 = double_expect(('peek', 'reverse_1'), probs)
+    A1B3 = double_expect(('peek', 'reverse_2'), probs)
+    A3B2 = double_expect(('reverse_2', 'reverse_1'), probs)
+    A3B3 = double_expect(('reverse_2', 'reverse_2'), probs)
```

Comment:
[nit]: (quote-style)

```suggestion
    A1B2 = double_expect(("peek", "reverse_1"), probs)
    A1B3 = double_expect(("peek", "reverse_2"), probs)
    A3B2 = double_expect(("reverse_2", "reverse_1"), probs)
    A3B3 = double_expect(("reverse_2", "reverse_2"), probs)
```